exporter/loadbalancingexporter/trace_exporter.go

```diff
@@ -207,20 +108,13 @@ func (e *traceExporterImp) consumeTrace(ctx context.Context, td pdata.Traces) er
 		return errNoTracesInBatch
 	}
 
-	// NOTE: make rolling updates of next tier of collectors work. currently this may cause
-	// data loss because the latest batches sent to outdated backend will never find their way out.
-	// for details: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1690
-	e.updateLock.RLock()
-	endpoint := e.ring.endpointFor(traceID)
-	exp, found := e.exporters[endpoint]
-	e.updateLock.RUnlock()
-	if !found {
-		// something is really wrong... how come we couldn't find the exporter??
-		return fmt.Errorf("couldn't find the exporter for the endpoint %q", endpoint)
+	exp, endpoint, err := e.loadBalancer.Exporter(traceID)
+	if err != nil {
+		return err
 	}
 
 	start := time.Now()
-	err := exp.ConsumeTraces(ctx, td)
+	err = exp.(component.TracesExporter).ConsumeTraces(ctx, td)
```

Comment:
What happens if the type isn't correct? Would it panic? If so, might be good to make a check before, even though the condition isn't possible with the current code.

Reply:
Added a check and error for this condition to prevent panic.

[`37c46e6` (#2633)](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2633/commits/37c46e611a2ccc626d1ea74d82e8220b9d85ad35)